black_it/calibrator.py

```diff
@@ -201,7 +214,7 @@ def set_samplers(self, samplers: List[BaseSampler]) -> None:
 
         """
         # overwrite the list of samplers
-        self.samplers = samplers
+        self.scheduler._samplers = tuple(samplers)  # pylint: disable=protected-access
```

Comment:
this should be illegal, and we inherit it from the not-so-clean-design of including `samplers_id_table` into calibrator. Anyway, topic for another PR, let's put this feature in first while keeping the rest to work as usual.